_posts/2016-05-12-no-more-redis.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Removing redis from our stack
+layout: post
+author: Arthur Nogueira Neves
+author_email: arthurnn@gmail.com
+---
+
+Hello Ruby community,
+
+Today we have some good news to announce: We no longer need Redis to run RubyGems.org. We have a brand new download stats system in place and we have shut down our Redis instances.
+
+## Is Redis bad?
+Not at all, if used correctly! A lot of companies use Redis, and there are a lot of use cases for it. One of the main use cases for Redis is to use it as a queue for background jobs. We use PostgreSQL and DelayedJob for our background jobs instead, so we don't need Redis for that. Our problems came from the way we were using it, and how much data we were storing. We were storing all the data in a permanent, never expiring fashion with quite a bit of storage and memory growth.
+The main problem we encountered with Redis, was that if an instance would go down, it would take many minutes to bring it up again, as it needs everything in memory. We had some Redis failures that took us about 45 minutes to bring it back up. This is not usually a problem when you don’t have a lot of data in Redis, like the background job case.
+
+## Our previous stats architecture with Redis
```

Comment:
What if we added architecture diagrams for these?

I generated the following with Graphviz. Feel free to tweak the names and boxes as you see fit.

```dot
digraph old {
  node [shape="box"];
  "User" [shape="ellipse"]
  rankdir=LR;
  "User" -> "Nginx" [label="request"];
  "Nginx" -> "RubyGems.org";
  "RubyGems.org" -> "Redis";
  "Redis" -> "RubyGems.org";
  "RubyGems.org" -> "Nginx";
  "Nginx" -> "User" [label="response"];
}
```
![old](https://cloud.githubusercontent.com/assets/10308/15227119/97342f52-183a-11e6-9ffc-23718e770f07.png)